src/queue/multilevel.rs

```diff
@@ -266,57 +286,67 @@ impl LevelManager {
         extras.schedule_time = Some(now());
     }
 
-    fn get_level0_chance(&self) -> u32 {
-        self.level0_chance.load(SeqCst)
-    }
-
     fn maybe_adjust_chance(&self) {
         if self.adjusting.compare_and_swap(false, true, SeqCst) {
             return;
         }
         // The statistics may be not so accurate because we cannot load two
         // atomics at the same time.
-        let total = self.total_elapsed.0.load(SeqCst);
-        if Duration::from_micros(total) < ADJUST_CHANCE_INTERVAL {
-            // Another thread just adjusted the chances.
+        let total = self.total_elapsed_us.get();
+        let total_diff = total - self.last_total_elapsed_us.get();
+        if total_diff < ADJUST_CHANCE_INTERVAL_US {
```

Comment:
So the metrics report interval should be less than `ADJUST_CHANCE_INTERVAL_US`, otherwise it can be always false.

Reply:
Sorry I don't get what you mean. Reporting the metrics shoudn't reset its value and `last_total_elapsed_us` is only set at L304.

Reply:
Oh, turns out I misunderstood the processing of metrics.